core/src/main/java/com/adyen/checkout/core/mbway/internal/ui/MBWayDelegate.kt

```diff
@@ -48,10 +61,29 @@ internal class MBWayDelegate(
             .stateIn(coroutineScope, SharingStarted.Lazily, stateManager.state.value.toViewState())
     }
 
-    override fun submit() = if (stateManager.isValid) {
-        // TODO Implement the submit logic
-    } else {
-        stateManager.highlightAllFieldValidationErrors()
+    override fun submit() {
+        if (stateManager.isValid) {
+            _eventFlow.tryEmit(
+                PaymentComponentEvent.Submit(
+                    // TODO - Correct data
```

Comment:
I would adjust the TODO here too
```suggestion
                    // TODO Adjust this logic when componentStateFlow is moved to here from `chore/state_management` branch.
```